### PR TITLE
chore(helm): added option to configure controlplane/ingress/egress dns

### PR DESCRIPTION
**Pull request by lukidzi**
### Checklist prior to review

Added an option in a helm values to configure DNS of controlPlane, ingress and egress pods

- [X] [Link to relevant issue][1] as well as docs and UI issues -- fix: https://github.com/kumahq/kuma/issues/10588
- [X] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Review by jijiechen — APPROVED** [review]

File: app/kumactl/cmd/install/testdata/install-control-plane.dump-values.yaml [hunk 1]
```diff
@@ -331,6 +331,16 @@ controlPlane:
   # -- If true, then control plane can support TLS secrets for builtin gateway outside of mesh system namespace.
   # The downside is that control plane requires permission to read Secrets in all namespaces.
   supportGatewaySecretsInAllNamespaces: false
+  # -- DNS configuration for the pod.
```
Comment (slonka) [inline app/kumactl/cmd/install/testdata/install-control-plane.dump-values.yaml:334]: for the CP pod, right? should we make it clear?

File: app/kumactl/cmd/install/testdata/install-control-plane.dump-values.yaml [hunk 2]
```diff
@@ -331,6 +331,16 @@ controlPlane:
...
+  dns:
+    # -- Defines how DNS resolution is configured for that Pod.
```
Comment (slonka) [inline app/kumactl/cmd/install/testdata/install-control-plane.dump-values.yaml:336]: should we state that this is equivalent to https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy and what values are available?

File: deployments/charts/kuma/templates/cp-deployment.yaml [hunk 3]
```diff
@@ -409,3 +409,21 @@ spec:
...
+        - {{ $nameserver }}
+        {{- end }}
+        {{- end }}
+        {{- if gt (len .Values.controlPlane.dns.config.searches) 0 }}
+        searches:
+        {{- range $search := .Values.controlPlane.dns.config.searches }}
+        - {{ $search }}
+        {{- end }}
+        {{- end }}
+      {{- end }}
```
Comment (slonka) [inline deployments/charts/kuma/templates/cp-deployment.yaml:429]: Could this be a function that we would call so we don't need to define this 3 times?

**Review by slonka — COMMENTED** [review]
Maybe we could improve the comments and dry up the implem a bit - otherwise 👍

**Review by jijiechen — APPROVED** [review]

**Review by slonka — APPROVED** [review]

**Pull request merged**